### PR TITLE
Javadoc and an infrastructure for the exception area (Artem)

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by Hanna-Kalashnyk — DISMISSED** [review]

File: src/main/java/com/exadel/discount/exception/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -14,60 +18,66 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import static com.exadel.discount.exception.type.ExceptionCause.*;
+import static org.apache.commons.lang3.StringUtils.upperCase;
```
Comment (el-dub) [inline src/main/java/com/exadel/discount/exception/GlobalExceptionHandler.java]: Remove wildcards

File: src/main/java/com/exadel/discount/exception/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -14,60 +18,66 @@
...
+
 @ControllerAdvice
 @Slf4j
 public class GlobalExceptionHandler {
+
     @ExceptionHandler(NotFoundException.class)
     @ResponseBody
     @ResponseStatus(HttpStatus.NOT_FOUND)
-    public ExceptionDetails handleNotFoundException(NotFoundException exception) {
+    public ExceptionDetails handleNotFoundValue(Exception exception) {
```
Comment (el-dub) [inline src/main/java/com/exadel/discount/exception/GlobalExceptionHandler.java]: I don't understand the aim of these changes and in other methotds as well

File: src/main/java/com/exadel/discount/exception/GlobalExceptionHandler.java (hunk 1 above)
Comment (ghost) [reply src/main/java/com/exadel/discount/exception/GlobalExceptionHandler.java]: :С

File: src/main/java/com/exadel/discount/exception/GlobalExceptionHandler.java (hunk 2 above)
Comment (ghost) [reply src/main/java/com/exadel/discount/exception/GlobalExceptionHandler.java]: :С

**Review by rudik-byte — DISMISSED** [review]

**Review by rudik-byte — DISMISSED** [review]

**Review by rudik-byte — DISMISSED** [review]

**Review by rudik-byte — DISMISSED** [review]

**Review by SergeyKoval — DISMISSED** [review]

**Review by rudik-byte — APPROVED** [review]

**Review by SergeyKoval — APPROVED** [review]

**Pull request merged**